app/src/main/java/com/jerboa/ui/components/blocks/BlocksActivity.kt

```diff
@@ -0,0 +1,242 @@
+                            }
+                        } else {
+                            items(
+                                items = personBlocks,
+                                contentType = { "personBlock" },
+                            ) { person ->
+                                val blockedPerson = person.target
+                                val id = blockedPerson.id
+                                val viewModel = makeBlockedPersonViewModel(id)
+
+                                BlockedElementListItem(
+                                    apiState = viewModel.blockPersonRes,
+                                    icon = blockedPerson.avatar,
+                                    name = blockedPerson.name,
+                                    onUnblock = { viewModel.blockPerson(false, context) },
+                                    onSuccessfulUnblock = {
+                                        personBlocks.removeIf { it.target.id == blockedPerson.id }
+                                    },
+                                )
+                            }
+                        }
+
+                        item { Spacer(modifier = Modifier.padding(vertical = MEDIUM_PADDING)) }
+                        item { Title(stringResource(R.string.blocked_communities)) }
+                        item { Spacer(modifier = Modifier.padding(vertical = SMALL_PADDING)) }
+
+                        if (communityBlocks.isNullOrEmpty()) {
+                            item {
+                                Text(stringResource(R.string.you_have_no_blocked_communities))
+                            }
+                        } else {
+                            items(
+                                items = communityBlocks,
+                                contentType = { "communityBlock" },
+                            ) { communityView ->
+                                val blockedCommunity = communityView.community
+                                val id = blockedCommunity.id
+                                val viewModel = makeBlockedCommunityViewModel(id)
+
+                                BlockedElementListItem(
+                                    apiState = viewModel.blockCommunityRes,
+                                    icon = blockedCommunity.icon,
+                                    name = blockedCommunity.name,
+                                    onUnblock = { viewModel.blockCommunity(false, context) },
+                                    onSuccessfulUnblock = {
+                                        communityBlocks.removeIf { it.community.id == blockedCommunity.id }
+                                    },
+                                )
+                            }
+                        }
+
+                        item { Spacer(modifier = Modifier.padding(vertical = MEDIUM_PADDING)) }
+                        item { Title(stringResource(R.string.blocked_instances)) }
```

Comment:
If you want https://developer.android.com/develop/ui/compose/lists#sticky-headers

Might look interesting for the subtitles